types/domain_type.go

```diff
@@ -0,0 +1,113 @@
+package types
+
+import (
+	"crypto/sha256"
+
+	"github.com/attestantio/go-eth2-client/spec/phase0"
+	"github.com/pkg/errors"
+)
+
+// NetworkID are intended to separate different SSV networks. A network can have many forks in it.
+type NetworkID [1]byte
+
+var (
+	MainnetNetworkID = NetworkID{0x0}
+	PrimusNetworkID  = NetworkID{0x1}
+	ShifuNetworkID   = NetworkID{0x2}
+	JatoNetworkID    = NetworkID{0x3}
+	JatoV2NetworkID  = NetworkID{0x4}
+)
+
+// DomainType is a unique identifier for signatures, 2 identical pieces of data signed with different domains will result in different sigs
+type DomainType [4]byte
+
+// DomainTypes represent specific forks for specific chains, messages are signed with the domain type making 2 messages from different domains incompatible
+var (
+	GenesisMainnet = DomainType{0x0, 0x0, MainnetNetworkID.Byte(), 0x0}
+	PrimusTestnet  = DomainType{0x0, 0x0, PrimusNetworkID.Byte(), 0x0}
+	ShifuTestnet   = DomainType{0x0, 0x0, ShifuNetworkID.Byte(), 0x0}
+	ShifuV2Testnet = DomainType{0x0, 0x0, ShifuNetworkID.Byte(), 0x1}
+	JatoTestnet    = DomainType{0x0, 0x0, JatoNetworkID.Byte(), 0x1}
+	JatoV2Testnet  = DomainType{0x0, 0x0, JatoV2NetworkID.Byte(), 0x1}
```

Comment:
There is an inconsistency in the fork version values assigned to `JatoTestnet` and `JatoV2Testnet`. Specifically, both have been assigned a fork version value of 0x1 instead of a sequential versioning like 0x0 and 0x1. I understand that this inconsistency emerged from an error when these networks were first introduced.

```go
JatoTestnet    = DomainType{0x0, 0x0, byte(JatoNetworkID), 0x1}
JatoV2Testnet  = DomainType{0x0, 0x0, byte(JatoV2NetworkID), 0x1}
```
To prevent confusion for individuals who might be reviewing or working with this code in the future, I suggest adding a comment in the code explaining this historical mistake. This comment could explain why JatoTestnet and JatoV2Testnet both have a fork version value of 0x1, and the historical reason behind it.

Suggested comment:
```go
// Historical Note:
// The fork version values for JatoTestnet and JatoV2Testnet are both set to 0x1 due to an error
// when these networks were initially introduced. This inconsistency does not align with the sequential
// versioning observed in other network and fork definitions. It's retained to maintain historical accuracy
// and to avoid any unforeseen issues that might arise from changing these established values.
// Future references and modifications should acknowledge this historical inconsistency.
var (
	GenesisMainnet = DomainType{0x0, 0x0, byte(MainnetNetworkID), 0x0}
	PrimusTestnet  = DomainType{0x0, 0x0, byte(PrimusNetworkID), 0x0}
	ShifuTestnet   = DomainType{0x0, 0x0, byte(ShifuNetworkID), 0x0}
	ShifuV2Testnet = DomainType{0x0, 0x0, byte(ShifuNetworkID), 0x1}
	JatoTestnet    = DomainType{0x0, 0x0, byte(JatoNetworkID), 0x1} // Note the fork version value
	JatoV2Testnet  = DomainType{0x0, 0x0, byte(JatoV2NetworkID), 0x1} // Note the fork version value
)
```

Reply:
I agree. Adding the comment. Thanks!